File: lecture_notes/10 refactoring/2-fix-style.js
'use strict';

const shift = (offset, points) => {
  const shifted = points.map((dot) => {
    const type = typeof dot;
    if (type === 'object') {
      dot.x += offset.x;
      dot.y += offset.y;
    } else {
      dot = JSON.parse(dot);
      dot.x += offset.x;
      dot.y += offset.y;
    }
    return dot;
  });
  return shifted;
};

const polyline = [
  { x: 0, y: 0 },
  { x: 10, y: 10 },
  '{ "x": 20, "y": 20 }',
  { x: 30, y: 30 },
];

const path = shift({ x: 10, y: -5 }, polyline);
console.log({ path });

File: lecture_notes/10 refactoring/2-fix-style.py
import json

def shift(offset, points):
    def add_move(point):
        type_1 = type(point)
        if type_1 == type({}):
            point['x'] += offset['x']
            point['y'] += offset['y']
        else:
            point = json.loads(point)
            point['x'] += offset['x']
            point['y'] += offset['y']
        return point
    moved = map(add_move, points)

    return moved



polyline = [
    { 'x': 0, 'y': 0 },
    { 'x': 10, 'y': 10 },
    '{ "x": 20, "y": 20 }',
    { 'x': 30, 'y': 30 },
]

path = shift({ 'x': 10, 'y': -5 }, polyline)
print(path)
